src/main/java/org/team1540/base/adjustables/AdjustableManager.java

```diff
@@ -0,0 +1,233 @@
+ * containing fields marked with {@link Tunable} or {@link Telemetry} to have those values show up
+ * on the SmartDashboard/Shuffleboard.
+ */
+public class AdjustableManager {
+
+  private static AdjustableManager
+      instance = new AdjustableManager();
+  private List<TunableField> tunables = new LinkedList<>();
+  private List<TelemetryField> telemetry = new LinkedList<>();
+
+  public static AdjustableManager getInstance() {
+    return instance;
+  }
+
+  /**
+   * Adds an object to the {@code AdjustableManager}.
+   *
+   * @param object The object to add. One or more of the fields in this object should be marked with
+   * {@link Telemetry} or {@link Tunable}.
+   */
+  public void add(Object object) {
+    // reflection time
+    Field[] fields = object.getClass().getFields();
+
+    boolean noneFound = true; // for logging to keep track if we have found at least one adjustable
+    for (Field field : fields) {
+
+      // process tunables
+      Tunable tunable = field.getAnnotation(Tunable.class);
+
+      if (tunable != null) {
+        // check if the field is of a supported type
+        TunableType tt = null;
+        for (TunableType type : TunableType
+            .values()) {
+          if (type.cls.isAssignableFrom(field.getType())) {
+            tt = type;
+            break;
+          }
+        }
+
+        if (tt == null) {
+          DriverStation.reportError(
+              "Annotated tunable in class added to AdjustableManager is not of a supported type",
+              false);
+          continue;
+        }
+
+        tunables.add(new TunableField(object, field, tt, tunable.value()));
+        noneFound = false;
+      }
+
+      // process telemetry
+      Telemetry teleAnnotation = field.getAnnotation(Telemetry.class);
+
+      if (teleAnnotation != null) {
+        // check if the field is of a supported type
+        TelemetryType tt = null;
+        for (TelemetryType type : TelemetryType
+            .values()) {
+          if (type.cls.isAssignableFrom(field.getType())) {
+            tt = type;
+            break;
+          }
+        }
+
+        if (tt == null) {
+          DriverStation.reportError(
+              "Annotated telemetry in class added to AdjustableManager is not of a supported type",
+              false);
+          continue;
+        }
+
+        telemetry.add(new TelemetryField(object, field, tt, teleAnnotation.value()));
+        noneFound = false;
+      }
+    }
+    if (noneFound) {
+      DriverStation.reportWarning(
+          "Object passed to AdjustableManager had no annotated adjustable fields",
+          false);
+    }
+  }
+
+  /**
+   * Updates adjustable values. This method should be called in {@code robotPeriodic()} in your main
+   * {@code Robot} class.
+   */
+  public void update() {
+    // Update tunables
+    for (TunableField tf : tunables) {
+      try {
+        if (SmartDashboard.containsKey(tf.label)) {
+          switch (tf.type) {
```

Comment:
I think there's a better way you can do this by fetching a reference to a method from the enums

Reply:
Store a lambda method in the enum and grab that